solver/llbsolver/solver.go

```diff
@@ -981,27 +981,20 @@ func loadEntitlements(b solver.Builder) (entitlements.Set, error) {
 }
 
 func loadSourcePolicy(b solver.Builder) (*spb.Policy, error) {
-	set := make(map[spb.Rule]struct{}, 0)
+	var srcPol spb.Policy
 	err := b.EachValue(context.TODO(), keySourcePolicy, func(v interface{}) error {
 		x, ok := v.(spb.Policy)
 		if !ok {
 			return errors.Errorf("invalid source policy %T", v)
 		}
 		for _, f := range x.Rules {
-			set[*f] = struct{}{}
+			r := *f
+			srcPol.Rules = append(srcPol.Rules, &r)
```

Comment:
I guess we should copy `Version` over as well?

Reply:
*sigh* and rules have pointers in them I guess we need to dereference.

Reply:
> sigh and rules have pointers in them I guess we need to dereference.

Do we need to make sure these are copied? I just left it like this because that is what old code did?

Reply:
This is probably fine since nothing is modifying the policies.

Reply:
I added Version just for completeness. Especially for the common case where `EachVertex` only returns one value. I think as a follow-up, returning the policy array would make more sense here.